rust-src/aggregate_sig/src/aggregate_sig.rs

```diff
@@ -171,6 +180,72 @@ pub fn verify_aggregate_sig<P: Pairing>(
     P::pair(&signature.0, &P::G2::one_point()) == product
 }
 
+/// Verifies an aggregate signature on pairs `(messages m_i, set_i)` `for
+/// i=1..n` but where `set_i` denotes the set of public keys corresponding to
+/// the secret keys that signed m_i. This implements a combination of AggregateVerify from Section 3.1.1 and FastAggregateVerify from Section 3.3.4 of https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature-05#section-3.1.1.
```

Comment:
The set is provided as a list. Are there are assumptions about duplicates? Please state them.

Reply:
More documentation added.